exercises/practice/dnd-character/tests/Tests.elm

```diff
@@ -0,0 +1,223 @@
+                test "ability modifier for score 15 is +2" <|
+                    \() ->
+                        DndCharacter.modifier 15
+                            |> Expect.equal 2
+            , skip <|
+                test "ability modifier for score 16 is +3" <|
+                    \() ->
+                        DndCharacter.modifier 16
+                            |> Expect.equal 3
+            , skip <|
+                test "ability modifier for score 17 is +3" <|
+                    \() ->
+                        DndCharacter.modifier 17
+                            |> Expect.equal 3
+            , skip <|
+                test "ability modifier for score 18 is +4" <|
+                    \() ->
+                        DndCharacter.modifier 18
+                            |> Expect.equal 4
+            ]
+        , describe "ability"
+            [ skip <|
+                fuzzWith { runs = 1000, distribution = Test.noDistribution }
+                    (Fuzz.fromGenerator DndCharacter.ability)
+                    "generated ability should be at least 3"
+                <|
+                    \ability -> Expect.atLeast 3 ability
+            , skip <|
+                fuzzWith { runs = 1000, distribution = Test.noDistribution }
+                    (Fuzz.fromGenerator DndCharacter.ability)
+                    "generated ability should be at most 18"
+                <|
+                    \ability -> Expect.atMost 18 ability
+            , skip <|
+                fuzzWith
+                    { runs = 10000
+                    , distribution =
+                        Test.expectDistribution
+                            [ ( Test.Distribution.moreThanZero, "is 3", (==) 3 ) ]
+                    }
+                    (Fuzz.fromGenerator DndCharacter.ability)
+                    "generated ability should sometimes be 3"
+                <|
+                    \_ -> Expect.pass
+            , skip <|
+                fuzzWith
+                    { runs = 1000
+                    , distribution =
+                        Test.expectDistribution
+                            [ ( Test.Distribution.moreThanZero, "is 18", (==) 18 ) ]
+                    }
+                    (Fuzz.fromGenerator DndCharacter.ability)
+                    "generated ability should sometimes be 18"
+                <|
+                    \_ -> Expect.pass
+            , skip <|
+                fuzzWith
+                    { runs = 10000
+                    , distribution =
+                        Test.expectDistribution
+                            [ ( Test.Distribution.atLeast 10, "is 13", (==) 13 )
+                            , ( Test.Distribution.atLeast 85, "is not 13", (/=) 13 )
+                            ]
+                    }
+                    (Fuzz.fromGenerator DndCharacter.ability)
+                    "13 has an approximate 13% chance of being picked"
+                <|
+                    \_ -> Expect.pass
```

Comment:
This one is also a test on the distribution. 

I wanted to test the actual probability of some abilities to appear. I picked 13 because it was the most common number, but I could add a couple more if you think it's worth doing.
The probability of 13 to appear is about 13.2%, so the probability that something else appears is 86.8%. When testing actual probabilities of events to happen, the docs say:
>Uses a statistical test to make sure the distribution doesn't pass or fail the distribution by accident (a flaky test). Will run more tests than specified with the runs config option if needed.
> 
> This has the consequence of running more tests the closer your expected distribution is to the true distribution. You can thus minimize and speed up this "making sure" process by requesting slightly less % of your distribution than needed.

So I used a window, anything between 10% and 15% (or actually, 85% of not having 13). Now again, for such a window, I needed to make sure to have enough runs so that 99.99% of the time it would work and I calculated the binomial distribution.
In simple terms, the binomial distribution can tell us the probability that an event will happen at least some percentage of the time, given the probability $p$ of the single event to happen and a number of runs $n$. Here are the numbers

<img width="413" alt="Screenshot 2023-10-07 at 22 07 01" src="https://github.com/exercism/elm/assets/16929078/df219a39-5db3-449e-907b-01d0efc2c821">

So in this case again we need 10k runs.

An error for a test of this type looks like
```
    Distribution report:
    ====================
      is not 13:  90.6%  (127810x)  ███████████████████████████░░░
      is 13:       9.4%   (13260x)  ███░░░░░░░░░░░░░░░░░░░░░░░░░░░
    
    Distribution of label "is 13" was insufficient:
      expected:  10.000%
      got:       9.400%.
    
    (Generated 141070 values.)
```
(that's quite a lot more values than specified, but it ran in 239 ms)